core/controllers/access_validators_test.py

```diff
@@ -563,3 +564,73 @@ def test_validation_returns_false_if_given_user_is_non_existent(
             ), expected_status_int=404
         )
         self.logout()
+
+
+class CollectionEditorPageAccessValidationHandler(test_utils.GenericTestBase):
+    """Checks the access to the collection editor page and its rendering."""
+
+    COLLECTION_ID: Final = '0'
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.collection_editor_username = 'collectionEditor'
+        self.guest_username = 'guest'
+        self.user_email = 'collectionEditor@example.com'
+        self.guest_email = 'guest@example.com'
+
+        system_user = user_services.get_system_user()
+        collection_services.load_demo(self.COLLECTION_ID)
+        rights_manager.release_ownership_of_collection(
+            system_user, self.COLLECTION_ID)
+
+        self.signup(self.user_email, self.collection_editor_username)
+        self.signup(self.guest_email, self.guest_username)
+        self.add_user_role(
+            self.collection_editor_username, feconf.ROLE_ID_COLLECTION_EDITOR)
+
+    def test_collection_editor_page_access_with_logging_in(self) -> None:
+        self.login(self.user_email)
+        self.get_html_response(
+            '%s/can_access_collection_editor_page/%s' % (
+            ACCESS_VALIDATION_HANDLER_PREFIX, self.COLLECTION_ID
+            ), expected_status_int=200
+        )
+        self.logout()
+
+    def test_collection_editor_page_access_after_logging_in(self) -> None:
+        self.login(self.user_email)
+        self.get_html_response(
+            '%s/can_access_collection_editor_page/%s' % (
+            ACCESS_VALIDATION_HANDLER_PREFIX, self.COLLECTION_ID
+            ), expected_status_int=200
+        )
+        self.logout()
+
+    def test_collection_editor_page_access_as_collection_editor(self) -> None:
+        self.login(self.user_email)
+        self.get_html_response(
+            '%s/can_access_collection_editor_page/%s' % (
+            ACCESS_VALIDATION_HANDLER_PREFIX, self.COLLECTION_ID
+            ), expected_status_int=200
+        )
+        self.logout()
+
+    def test_validation_returns_true_for_guest_user(
```

Comment:
Why `true` for guest users? It's should raise 401, right?

Reply:
I updated the test cases, PTAL